versions-common/src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -479,13 +465,30 @@ private List<IgnoreVersion> getIgnoredVersions( Artifact artifact )
 
     @Override
     public void resolveArtifact( Artifact artifact, boolean usePluginRepositories )
-        throws ArtifactResolutionException, ArtifactNotFoundException
+        throws ArtifactResolutionException
     {
-        artifactResolver.resolve( artifact,
-                usePluginRepositories
-                        ? mavenSession.getCurrentProject().getPluginArtifactRepositories()
-                        : mavenSession.getCurrentProject().getRemoteArtifactRepositories(),
-                localRepository );
+        try
+        {
+            ArtifactResult artifactResult = aetherRepositorySystem.resolveArtifact( mavenSession.getRepositorySession(),
+                    new ArtifactRequest( toArtifact( artifact ),
+                            usePluginRepositories
+                                    ? mavenSession.getCurrentProject().getRemotePluginRepositories()
+                                    : mavenSession.getCurrentProject().getRemoteProjectRepositories(),
+                            getClass().getName() ) );
+            artifact.setFile( artifactResult.getArtifact().getFile() );
+            artifact.setVersion( artifactResult.getArtifact().getVersion() );
+            artifact.setResolved( artifactResult.isResolved() );
+            if ( !artifactResult.getExceptions().isEmpty() )
+            {
+                getLog().warn( "Exceptions thrown while trying to resolve artifact " + artifact );
+                artifactResult.getExceptions().forEach( e ->
+                        getLog().warn( "\t" + e.getMessage() ) );
+            }
```

Comment:
I think that never occurs in this place. 

When exceptions list is set also exception is thrown.

Reply:
It looks like you're right. Removed.